project/stac_api/models.py

```diff
@@ -1,28 +1,307 @@
+    # array field of CharFields. Simple validation is done (e.g. no "Sonderzeichen"
+    # in array)
+    geoadmin_variant = ArrayField(models.CharField())
+    proj = models.IntegerField(null=True)
+    title = models.CharField()
+    media_type = models.CharField()
+    copyFromHref = models.URLField()
+    location = models.URLField()
+
+    def __str__(self):
+        return self.asset_name
+
+    def clean(self):
+        # very simple validation, raises error when geoadmin_variant strings contain special
+        # characters or umlaut.
+        for variant in self.geoadmin_variant:
+            if not (bool(re.search('^[a-zA-Z0-9]*$', variant)) == True):
+                raise ValidationError(_('Property geoadmin:variant not correctly specified.'))
+
+
+class Keyword(models.Model):
+    collections = models.ManyToManyField(Collection)
+    keyword = models.TextField(blank=False)  # string
+
+
+# TODO: probably provider on collection level could be auto-populated and updated
+# on every update of an item inside the collection
+class Provider_Collection(models.Model):
+    collections = models.ManyToManyField(Collection)
+    name = models.TextField(blank=False)  # string
+    description = models.TextField()  # string
+    roles = models.TextField()  # [string]
+    # possible roles are licensor, producer, processor or host. Probably it might sense
+    url = models.URLField()  # string
+
+    def clean(self):
+        allowed_roles = ['licensor', 'producer', 'processor', 'host']
+        for role in self.roles:
+            if role not in allowed_roles:
+                raise ValidationError(_('Incorrectly defined role found'))
+
+
+class Provider_Item(models.Model):
+    items = models.ManyToManyField(Item)
+    name = models.TextField(blank=False)  # string
+    description = models.TextField()  # string
+    roles = models.TextField()  # [string]
+    # possible roles are licensor, producer, processor or host. Probably it might sense
+    url = models.URLField()  # string
+
+    def clean(self):
+        allowed_roles = ['licensor', 'producer', 'processor', 'host']
+        for role in self.roles:
+            if role not in allowed_roles:
+                raise ValidationError(_('Incorrectly defined role found'))
+
+
+class Link_Collection(models.Model):
+    collections = models.ManyToManyField(Collection)
+    href = models.URLField()  # string
+    rel = models.CharField()  # string
+    link_type = models.CharField(blank=True)  # string
+    # added link_ to the fieldname, as "type" is reserved
+    title = models.CharField(blank=True)  # string
+
+
+class Link_Item(models.Model):
+    items = models.ManyToManyField(Item)
+    href = models.URLField()  # string
+    rel = models.CharField()  # string
+    link_type = models.CharField(blank=True)  # string
+    # added link_ to the fieldname, as "type" is reserved
+    title = models.CharField(blank=True)  # string
```

Comment:
see comment above about M2M, 
I would just define one model `Link`, and have M2M fields for `links` in `Collection` and `Item`

Reply:
Adapted that way